src/graph.cpp

```diff
@@ -242,21 +243,42 @@ std::vector<std::set<node_id_t>> Graph::boruvka_emulation(bool make_copy) {
       copy_supernodes[i] = nullptr;
   }
 
-  do {
-    modified = false;
-    sample_supernodes(query, reps);
-    std::vector<std::vector<node_id_t>> to_merge = supernodes_to_merge(query, reps);
-
-    // make a copy if necessary
-    if(make_copy && first_round) {
-      backed_up = reps;
-      if (!copy_in_mem) backup_to_disk(backed_up);
+  // function to restore supernodes after CC if make_copy is specified
+  auto cleanup_copy = [&make_copy, this, &backed_up, &copy_supernodes]() {
```

Comment:
This code looks right. I'm curious if you've tried inducing an exception to ensure the super-nodes are restored correctly. Maybe this should be a unit test?

Reply:
Yes, you’re right. To do so we’ll have to add a dependency that allows us to mock. I can work on that if you think it’s important.

Reply:
I'll add it to the Trello and either you or someone else can be assigned to it. Doesn't have to be a part of this PR.